app.js

```diff
@@ -246,3 +262,17 @@ app.get('/', function(req, res) {
 });
 
 
+function infiniteLoop() {
+    /* this will launch other scheduled tasks too */
+    return Promise
+        .resolve()
+        .delay(60 * 1000)
+        .then(function() {
+            if(_.size(performa.queue))
+                return mongo
+                    .cacheFlush(performa.queue, "performa")
+        })
+        .then(infiniteLoop);
```

Comment:
I was using it, https://github.com/tracking-exposed/facebook/commit/4576ae42c4243951ef6fdbfe428dda8fd65c7a4d but I experienced some deadlock, that one is a pattern I saw from the official stackoverflow conversation [ http://stackoverflow.com/questions/33222421/what-is-a-good-pattern-for-interval-with-timeout-using-promises ]. I looked after it because I saw, the mongo operation stop working after the first call lead by `setInterval`